packages/garbo/src/tasks/cockroachPrep.ts

```diff
@@ -16,35 +15,84 @@ import {
 import {
   abort,
   adv1,
+  Effect,
+  effectModifier,
+  isShruggable,
+  Item,
   mallPrice,
   myBuffedstat,
+  myEffects,
+  retrieveItem,
   runChoice,
+  Stat,
+  toEffect,
+  use,
   visitUrl,
 } from "kolmafia";
 import { garboValue } from "../garboValue";
 import { freeFightFamiliar } from "../familiar";
-import { freeFightOutfit } from "../outfit";
+import { freeFightOutfit, meatTargetOutfit } from "../outfit";
 import { GarboStrategy, Macro } from "../combat";
 import { acquire } from "../acquire";
 
+function pickBestDebuff(stat: Stat): Item {
+  const statName = stat.toString();
+
+  const debuffMenu = Item.all()
+    .filter((it) => {
+      const effect = effectModifier(it, "Effect").toString();
+      return effect && !(effect in myEffects());
+    })
+    .filter((it) => getModifier(statName, it) < 0)
+    .map(
+      (it) => [it, mallPrice(it) / getModifier(statName, it)] as [Item, number],
+    ); // getModifier should return a negative value, flipping these negative
+
+  // ...so that when we maxBy we pick the smallest negative value, spending the least meat
+  return maxBy(debuffMenu, ([, value]) => value)[0];
+}
```

Comment:
function getBestDebuffItem(stat: Stat): Item {
  const debuffs = Item.all()
    .map((item) => ({ item, effect: effectModifier(it, "Effect") })
    .filter(({ effect }) => effect !== $effect.none && !have(effect) && getModifier(stat.toString(), effect) < 0);
    
  return maxBy(debuffs, ({ item, effect }) => mallPrice(item) / getModifier(stat.toString(), effect)).item;
}

This still has the issue of ignoring the various `Percent` buffs, but I suppose those are unlikely to matter much when wearing the eyepatch

Reply:
I think we test it and if perfect buffs matter, reassess.